lib/active_record_shards/migration.rb

```diff
@@ -1,25 +1,32 @@
 # frozen_string_literal: true
 module ActiveRecord
   class Migrator
-    class << self
-      [:up, :down, :run].each do |m|
-        define_method("#{m}_with_sharding") do |*args|
-          ActiveRecord::Base.on_shard(nil) do
-            send("#{m}_without_sharding", *args)
-          end
-          ActiveRecord::Base.on_all_shards do
-            send("#{m}_without_sharding", *args)
-          end
-        end
-        alias_method :"#{m}_without_sharding", m.to_sym
-        alias_method m.to_sym, :"#{m}_with_sharding"
+    def self.migration_context
```

Comment:
if we made that method up, then prefer prefixing `shards_` or so to make that obvious